### PR TITLE
ptCollage Style & Stylesheet framework

### PR DESCRIPTION
**Pull request by EwanGreen4**
This introduces a replica ptCollage style into ptCollab. It looks like this:

image

It is as accurate to ptCollage as I could get without manually setting fixed widths/heights or otherwise intrusive behavior.

More importantly, though, it's introduced a method of importing & parsing stylesheets from a file. Currently, the location of these styles is in `/src/style/`, but `/style/` should be moved to the location of the binary at compile-time (i.e. `/build/style/`).
Each custom style should get its own folder within the style directory, so that they can be chosen in the settings menu. The stylsheet file should be named the same as the directory.
For example:
`ptcollab-dir/style/CustomStyle/CustomStyle.qss`

### CODE REVIEW

**Comment by EwanGreen4** [comment]
Update:
- Stylesheet files are now just .qss (variable system has been discarded)
- Palette files are (optionally) represented in the style's palette.ini. If they aren't present, it will use the default palette
- Style selection menu has been implemented, CustomStyle QSettings boolean has been removed in favor of a StyleName string that is "Default" by default. If the setting is "Default", no custom style will be applied (it is also shown in the selection combo box)

**Comment by EwanGreen4** [comment]
fixed a bug where available style names were reapplied

File: src/editor/StyleEditor.cpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                  // their respected style is not problematic. The name of this
+                  // style should be consistent, reliable, and guaranteed on a
+                  // fresh instance (installation or compilation). And if not,
+                  // there won't be any errors.
+
+void determineStyle() {
+  QString styleName = Settings::StyleName::get();
+  if (styleName == "System") return;
+  if (styleName.isEmpty() ||
+      !QFile::exists(qApp->applicationDirPath() + "/style/" + styleName + "/" +
```
Comment (yuxshao) [inline src/editor/StyleEditor.cpp:30]: Would you mind refactoring this logic that gets the qss file of a stylesheet into a function maybe?

Comment (EwanGreen4) [reply src/editor/StyleEditor.cpp]: done

**Pull request merged**